lib/join/server.go

```diff
@@ -143,29 +148,48 @@ func (s *Server) Join(stream messages.ServerStream) (err error) {
 		return trace.Wrap(err)
 	}
 
-	// Fetch the provision token and validate that it is not expired.
-	provisionToken, err := s.cfg.AuthService.ValidateToken(ctx, clientInit.TokenName)
+	var token provision.Token
+	scopedTokenRes, err := s.cfg.ScopedTokenService.GetScopedToken(ctx, &joiningv1.GetScopedTokenRequest{
+		Name: clientInit.TokenName,
+	})
 	if err != nil {
-		return trace.Wrap(err)
+		if !trace.IsNotFound(err) {
+			log.ErrorContext(ctx, "failed to fetch scoped token", "error", err)
+		}
+
+		// Fetch the provision token and validate that it is not expired.
+		provisionToken, err := s.cfg.AuthService.ValidateToken(ctx, clientInit.TokenName)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		token = provisionToken
 	}
+
+	if token == nil {
+		token, err = scopedjoining.NewToken(scopedTokenRes.GetToken())
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
```

Comment:
can this get pulled into a single method that returns an appropriate validated token given a token name?